presto-main/src/main/java/com/facebook/presto/execution/scheduler/group/DynamicBucketNodeMap.java

```diff
@@ -30,30 +32,45 @@
         extends BucketNodeMap
 {
     private final int bucketCount;
-    private final Int2ObjectMap<InternalNode> bucketToNode = new Int2ObjectOpenHashMap<>();
+    private final Int2ObjectMap<InternalNodeInfo> bucketToNodeInfo = new Int2ObjectOpenHashMap<>();
+    private final boolean hasInitialMap;
 
     public DynamicBucketNodeMap(ToIntFunction<Split> splitToBucket, int bucketCount)
     {
         super(splitToBucket);
         checkArgument(bucketCount > 0, "bucketCount must be positive");
         this.bucketCount = bucketCount;
+        hasInitialMap = false;
     }
 
-    public DynamicBucketNodeMap(ToIntFunction<Split> splitToBucket, int bucketCount, List<InternalNode> bucketToPreferredNode)
+    public DynamicBucketNodeMap(ToIntFunction<Split> splitToBucket, int bucketCount, List<InternalNode> bucketToPreferredNode, boolean cacheable)
```

Comment:
ditto, I'm not sure if this `cacheable` is useful. Bucket is initialized to preferred node, so they should be cacheable?